lib/ansiblelint/__init__.py

```diff
@@ -259,7 +259,11 @@ def run(self):
                 continue
             if playbook[1] == 'role':
                 continue
-            files.append({'path': ansiblelint.utils.normpath(playbook[0]), 'type': playbook[1]})
+            files.append({'path': ansiblelint.utils.normpath(playbook[0]),
+                          'type': playbook[1],
+                          # add an absolute path here, so rules are able to validate if
+                          # referenced files exist
+                          'abs_path': os.path.dirname(playbook[0])})
```

Comment:
this name is weird: it looks like it's a dir path but the implication seems to be that it's a file path.

Reply:
would `absolute_directory` be a better name?

Reply:
probably. but I'm still not sure if it's the right design decision

Reply:
fair enough. i did not know where this information was available for rules so i put it there.